pkg/objstore/oss/oss.go

```diff
@@ -0,0 +1,349 @@
+// Package OSS implements common object storage abstractions against Alibaba object storage service(OSS).
+package oss
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	slog "log"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/improbable-eng/thanos/pkg/runutil"
+
+	"github.com/aliyun/aliyun-oss-go-sdk/oss"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/improbable-eng/thanos/pkg/objstore"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+// DirDelim is the delimiter used to model a directory structure in an object store bucket.
+const DirDelim = "/"
+
+// Config stores the configuration for OSS bucket.
+type Config struct {
+	Bucket     string     `yaml:"bucket"`
+	Endpoint   string     `yaml:"endpoint"`
+	AccessKey  string     `yaml:"access_key"`
+	SecretKey  string     `yaml:"secret_key"`
+	HTTPConfig HTTPConfig `yaml:"http_config"`
+	LogLevel   int        `yaml:"log_level"`
```

Comment:
Would it print out oss log in thanos? IMO the error messages should be returned from oss package, and we can catch and throw it to the user. I don't know what do you concern for the oss logger. But IMO it is unnecessary, we should have a unified logger handler. What do you think?